realm-flipper-plugin-device/src/ConvertFunctions.tsx

```diff
@@ -4,43 +4,47 @@ import {
   CanonicalObjectSchema,
   CanonicalObjectSchemaProperty,
   Object as RealmObject,
-} from "realm";
+  ObjectSchema,
+} from 'realm';
 
 type PropertyDescription = {
   type: string;
   objectType?: string;
 };
 
+// TODO: this function can probably be simplified as it largely just
+// serializes object.toJSON() which supports circular relationships now.
 const convertObjectToDesktop = (
   object: RealmObject,
-  properties: {
-    [keys: string]: PropertyDescription;
-  }
+  properties: Realm.PropertiesTypes,
 ) => {
-  const obj = {};
-  Object.keys(properties).forEach((key) => {
+  const obj = Object();
+  Object.keys(properties).forEach(key => {
+    const jsonifiedObject = object.toJSON();
     const property = properties[key] as PropertyDescription;
     // make a copy of the object
-    obj[key] = object.toJSON()[key];
+    obj[key] = jsonifiedObject[key];
 
-    if (property.type === "object" && obj[key]) {
+    if (property.type === 'object' && obj[key]) {
+      //@ts-expect-error We know the key exists
       const objectKey = object[key]._objectKey();
-      obj[key]._objectKey = objectKey;
+      // Store object key as a seperate key for the plugin
+      obj[key]._pluginObjectKey = objectKey;
     }
   });
-  const replacer = (key, value) => {
+  const replacer = (key: any, value: any) => {
```

Comment:
Could constrain these a bit more.  
```suggestion
  const replacer = (key: string, value: unknown) => {
```

Reply:
Perhaps this would be even better: `key: keyof typeof properties`